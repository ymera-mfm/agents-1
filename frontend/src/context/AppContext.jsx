// AppContext wrapper
// Re-exports AppContext from store for backward compatibility

export { AppProvider, useApp } from '../store/AppContext';
export * from '../store/AppContext';
