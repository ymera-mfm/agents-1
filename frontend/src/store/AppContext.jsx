import React, { createContext, useContext, useState, useMemo, useCallback } from 'react';
import { apiService } from '../services/api';
import { websocketService } from '../services/websocket';
import { cacheService } from '../services/cache';

const AppContext = createContext(null);

export const useApp = () => {
  const context = useContext(AppContext);
  if (!context) {
    throw new Error('useApp must be used within AppProvider');
  }
  return context;
};

export const AppProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [page, setPage] = useState('login');
  const [agents, setAgents] = useState([]);
  const [projects, setProjects] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [notifications, setNotifications] = useState([]);
  const [collaborationSessions, setCollaborationSessions] = useState([]);
  const [resourceAllocations, setResourceAllocations] = useState({});

  const loadData = useCallback(async () => {
    try {
      setLoading(true);

      // Mock data for demo
      const agentsData = [
        {
          id: 1,
          name: 'DataMiner Pro',
          status: 'working',
          type: 'data',
          efficiency: 94,
          tasks: 23,
          color: '#00f5ff',
        },
        {
          id: 2,
          name: 'CodeWeaver',
          status: 'thinking',
          type: 'dev',
          efficiency: 88,
          tasks: 15,
          color: '#ff3366',
        },
        {
          id: 3,
          name: 'CloudOrchestrator',
          status: 'idle',
          type: 'cloud',
          efficiency: 91,
          tasks: 8,
          color: '#00ff88',
        },
        {
          id: 4,
          name: 'SecurityGuard',
          status: 'working',
          type: 'security',
          efficiency: 96,
          tasks: 31,
          color: '#ffd700',
        },
      ];

      const projectsData = [
        {
          id: 1,
          name: 'E-Commerce Platform',
          description: 'Building a scalable e-commerce solution',
          status: 'in_progress',
          progress: 67,
          assignedAgents: [1, 2],
          requiredAgents: 2,
          team: 5,
          budget: 100000,
          spent: 67000,
          milestones: [
            {
              id: 1,
              name: 'Requirements Gathering',
              duration: 14,
              startDate: '2024-01-01',
              progress: 100,
              completed: true,
            },
            {
              id: 2,
              name: 'UI/UX Design',
              duration: 21,
              startDate: '2024-01-15',
              progress: 100,
              completed: true,
            },
            {
              id: 3,
              name: 'Backend Development',
              duration: 35,
              startDate: '2024-02-05',
              progress: 75,
              completed: false,
            },
          ],
          phases: [
            { id: 1, name: 'Analysis', status: 'completed', progress: 100, color: '#10b981' },
            { id: 2, name: 'Design', status: 'completed', progress: 100, color: '#3b82f6' },
            { id: 3, name: 'Development', status: 'in_progress', progress: 60, color: '#f59e0b' },
          ],
        },
      ];

      setAgents(agentsData);
      setProjects(projectsData);

      cacheService.set('agents', agentsData);
      cacheService.set('projects', projectsData);
    } catch (err) {
      setError(err.message);
      console.error('Failed to load data:', err);
    } finally {
      setLoading(false);
    }
  }, []);

  const login = useCallback(
    async (username, _password) => {
      try {
        setLoading(true);
        setError(null);

        // Mock login for demo
        const mockUser = {
          id: Date.now(),
          name: username,
          avatar: username[0].toUpperCase(),
          email: `${username}@agentflow.com`,
          bio: 'This is a demo user profile for the AgentFlow application.',
        };

        setUser(mockUser);
        setPage('dashboard');
        await loadData();
      } catch (err) {
        setError(err.message);
        throw err;
      } finally {
        setLoading(false);
      }
    },
    [loadData]
  );

  const logout = useCallback(() => {
    setUser(null);
    setPage('login');
    apiService.setToken(null);
    websocketService.disconnect();
    cacheService.clear();
    setAgents([]);
    setProjects([]);
  }, []);

  const updateAgentStatus = useCallback((id, status) => {
    setAgents((prev) => prev.map((agent) => (agent.id === id ? { ...agent, status } : agent)));
  }, []);

  const handleResourceAllocation = useCallback((allocations) => {
    setResourceAllocations(allocations);
    setProjects((prev) =>
      prev.map((project) => ({
        ...project,
        assignedAgents: allocations[project.id] || project.assignedAgents,
      }))
    );
  }, []);

  const addCollaborationSession = useCallback((session) => {
    setCollaborationSessions((prev) => [...prev, session]);
  }, []);

  const value = useMemo(
    () => ({
      user,
      login,
      logout,
      page,
      setPage,
      agents,
      setAgents,
      projects,
      setProjects,
      loading,
      error,
      notifications,
      setNotifications,
      collaborationSessions,
      addCollaborationSession,
      resourceAllocations,
      handleResourceAllocation,
      updateAgentStatus,
    }),
    [
      user,
      page,
      agents,
      projects,
      loading,
      error,
      notifications,
      collaborationSessions,
      resourceAllocations,
      login,
      logout,
      addCollaborationSession,
      handleResourceAllocation,
      updateAgentStatus,
    ]
  );

  return <AppContext.Provider value={value}>{children}</AppContext.Provider>;
};
