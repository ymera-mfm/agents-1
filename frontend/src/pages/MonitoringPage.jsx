import React, { useMemo } from 'react';
import { useApp } from '../store/AppContext';
import { Cpu, Folder } from 'lucide-react';

export const MonitoringPage = () => {
  const { agents, projects } = useApp();

  const monitoringData = useMemo(() => {
    return {
      agents: agents.map((agent) => ({
        ...agent,
        statusColor:
          agent.status === 'working'
            ? '#10b981'
            : agent.status === 'thinking'
              ? '#f59e0b'
              : '#6b7280',
      })),
      projects: projects.map((project) => ({
        ...project,
        statusColor:
          project.status === 'in_progress'
            ? '#3b82f6'
            : project.status === 'completed'
              ? '#10b981'
              : '#a855f7',
      })),
    };
  }, [agents, projects]);

  return (
    <div className="min-h-screen pt-24 pb-12 px-4 sm:px-6 lg:px-8">
      <div className="max-w-7xl mx-auto space-y-8">
        <div>
          <h1 className="text-4xl font-bold text-white mb-2">Monitoring</h1>
          <p className="text-gray-400">Real-time monitoring of agents and projects</p>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
          <div className="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6">
            <h3 className="text-xl font-bold text-white mb-6">Agent Status</h3>
            <div className="space-y-3">
              {monitoringData.agents.map((agent) => (
                <div
                  key={agent.id}
                  className="flex items-center justify-between p-4 bg-white/5 rounded-lg hover:bg-white/10 transition"
                >
                  <div className="flex items-center space-x-3">
                    <div
                      className="w-10 h-10 rounded-full flex items-center justify-center"
                      style={{ backgroundColor: `${agent.color}20` }}
                    >
                      <Cpu className="w-5 h-5" style={{ color: agent.color }} />
                    </div>
                    <div>
                      <h4 className="font-semibold text-white">{agent.name}</h4>
                      <p className="text-sm text-gray-400">
                        {agent.tasks} tasks â€¢ {agent.efficiency}% efficiency
                      </p>
                    </div>
                  </div>
                  <span
                    className="text-xs px-3 py-1 rounded-full capitalize"
                    style={{ backgroundColor: `${agent.statusColor}20`, color: agent.statusColor }}
                  >
                    {agent.status}
                  </span>
                </div>
              ))}
            </div>
          </div>

          <div className="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6">
            <h3 className="text-xl font-bold text-white mb-6">Project Status</h3>
            <div className="space-y-3">
              {monitoringData.projects.map((project) => (
                <div
                  key={project.id}
                  className="flex items-center justify-between p-4 bg-white/5 rounded-lg hover:bg-white/10 transition"
                >
                  <div className="flex items-center space-x-3">
                    <div
                      className="w-10 h-10 rounded-lg flex items-center justify-center"
                      style={{ backgroundColor: `${project.statusColor}20` }}
                    >
                      <Folder className="w-5 h-5" style={{ color: project.statusColor }} />
                    </div>
                    <div>
                      <h4 className="font-semibold text-white">{project.name}</h4>
                      <p className="text-sm text-gray-400">{project.progress}% complete</p>
                    </div>
                  </div>
                  <span
                    className="text-xs px-3 py-1 rounded-full capitalize"
                    style={{
                      backgroundColor: `${project.statusColor}20`,
                      color: project.statusColor,
                    }}
                  >
                    {project.status.replace('_', ' ')}
                  </span>
                </div>
              ))}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MonitoringPage;
