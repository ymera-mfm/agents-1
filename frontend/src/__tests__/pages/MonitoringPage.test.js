import React from 'react';
import { render } from '@testing-library/react';
import { AppProvider } from '../../context/AppContext';
import MonitoringPage from '../../pages/MonitoringPage';

describe('MonitoringPage', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <AppProvider>
        <MonitoringPage />
      </AppProvider>
    );
    expect(container).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <AppProvider>
        <MonitoringPage />
      </AppProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
