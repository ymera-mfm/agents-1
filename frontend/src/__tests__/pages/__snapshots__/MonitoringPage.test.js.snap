// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitoringPage matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen pt-24 pb-12 px-4 sm:px-6 lg:px-8"
  >
    <div
      class="max-w-7xl mx-auto space-y-8"
    >
      <div>
        <h1
          class="text-4xl font-bold text-white mb-2"
        >
          Monitoring
        </h1>
        <p
          class="text-gray-400"
        >
          Real-time monitoring of agents and projects
        </p>
      </div>
      <div
        class="grid grid-cols-1 lg:grid-cols-2 gap-6"
      >
        <div
          class="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6"
        >
          <h3
            class="text-xl font-bold text-white mb-6"
          >
            Agent Status
          </h3>
          <div
            class="space-y-3"
          />
        </div>
        <div
          class="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6"
        >
          <h3
            class="text-xl font-bold text-white mb-6"
          >
            Project Status
          </h3>
          <div
            class="space-y-3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
